client/src/app/products/product.ts

```diff
@@ -14,7 +14,36 @@ export interface Product {
   image?: string; // server filter
 }
 
-// eslint-disable-next-line max-len
-export type ProductCategory = 'baked goods' | 'produce' | 'meat' | 'dairy' | 'frozen foods' | 'baking supplies'
-| 'beverages' | 'cleaning products' | 'miscellaneous' | 'deli'
-|'herbs and spices' | 'paper products' | 'pet supplies' | 'staples' | 'toiletries';
+export const categories =
+  ['baked goods', 'produce', 'meat', 'dairy', 'frozen foods',
+   'baking supplies', 'beverages', 'cleaning products', 'miscellaneous', 'deli',
+   'herbs and spices', 'paper products', 'pet supplies', 'staples', 'toiletries'] as const;
+export type ProductCategory = typeof categories[number];
+
+export const categoryCamelCase = (category: ProductCategory) => {
+  // This is essentially a hack to handle all the categories with spaces
+  // in them. One could have done that programmatically, but I didn't want
+  // to add that computational overhead given how often this will be called.
+  // I suspect there's a better solution, TBH.
+  const categoryMap = {
+    'baked goods': 'bakedGoods',
```

Comment:
I could _totally_ have a typo here in this `categoryMap`, so you probably want to double check it for me.